tests/ef_tests.rs

```diff
@@ -194,15 +193,30 @@ fn run_test(path: &Path, contents: String) -> datatest_stable::Result<()> {
 
             let res = evm.transact().unwrap();
 
-            if test.expect_exception.is_some() {
-                assert!(!res.result.is_success());
-                // NOTE: the expect_exception string is an error description, we don't check the expected error
-                continue;
+            match (&test.expect_exception, &res.result) {
+                (None, _) => {
+                    if let Some((expected_output, output)) =
+                        unit.out.as_ref().zip(res.result.output())
+                    {
+                        if expected_output != output {
```

Comment:
Isn't this ignoring output mismatches?
I think we can simplify this to:

```suggestion
                    if unit.out.as_ref() != res.result.output() {
```